### PR TITLE
Static pr creator

### PR DESCRIPTION
**Pull request by WChrisK**

### CODE REVIEW

**Comment by WChrisK** [comment]
I want to make more unit tests (mainly for the CSV), figured I'd get this up first though. There's a few things I need to investigate however.

**Comment by WChrisK** [comment]
When I test this on my local machine its fine:
```Finished in 14.89 seconds (files took 2.56 seconds to load)
31 examples, 0 failures
```

File: app/models/peer_review.rb [hunk 1]
```diff
@@ -29,4 +29,30 @@ def no_students_should_be_reviewer_and_reviewee
       end
     end
   end
+
+  # Gets the peer review between the groups, or returns nill if not found.
+  def self.get_peer_review_for(reviewer, reviewee)
+    results = Result.where(submission: reviewee.current_submission_used)
+    PeerReview.where(result: results, reviewer: reviewer).first
```
Comment (david-yz-liu) [inline app/models/peer_review.rb]: You can use `find_by` here

File: app/models/peer_review.rb [hunk 2]
```diff
@@ -29,4 +29,30 @@ def no_students_should_be_reviewer_and_reviewee
...
+  end
+
+  def self.does_review_exist_between?(reviewer, reviewee)
```
Comment (david-yz-liu) [inline app/models/peer_review.rb]: No need to use `does` - `review_exists_between?`

File: app/models/peer_review.rb [hunk 3]
```diff
@@ -29,4 +29,30 @@ def no_students_should_be_reviewer_and_reviewee
...
+    not PeerReview.get_peer_review_for(reviewer, reviewee).nil?
+  end
+
+  def self.able_to_assign_peer_review_to?(reviewer, reviewee)
```
Comment (david-yz-liu) [inline app/models/peer_review.rb]: I think I've seen `can` rather than `able_to` more consistently

File: app/models/peer_review.rb [hunk 4]
```diff
@@ -29,4 +29,30 @@ def no_students_should_be_reviewer_and_reviewee
...
+
+  # Creates a new peer review between the reviewer and reviewee, otherwise if
+  # one exists it returns nil
+  def self.create_peer_review_between(reviewer, reviewee)
+    if PeerReview.able_to_assign_peer_review_to?(reviewer, reviewee)
+      result = Result.create!(submission: reviewee.current_submission_used,
+                              marking_state: Result::MARKING_STATES[:incomplete])
+      return PeerReview.create!(reviewer: reviewer, result: result)
+    end
+    nil
```
Comment (david-yz-liu) [inline app/models/peer_review.rb]: It seems this `nil` is unnecessary

File: app/models/peer_review.rb [hunk 5]
```diff
@@ -29,4 +29,30 @@ def no_students_should_be_reviewer_and_reviewee
...
+  def self.does_review_exist_between?(reviewer, reviewee)
+    not PeerReview.get_peer_review_for(reviewer, reviewee).nil?
+  end
+
+  def self.able_to_assign_peer_review_to?(reviewer, reviewee)
+    return (not PeerReview.does_review_exist_between?(reviewer, reviewee)) &&
+        reviewer.does_not_share_any_students?(reviewee)
+  end
+
+  # Creates a new peer review between the reviewer and reviewee, otherwise if
```
Comment (david-yz-liu) [inline app/models/peer_review.rb]: It would be helpful to mention that `reviewer` and `reviewee` are both groupings.

**Comment by david-yz-liu** [comment]
@WChrisK this is exactly what I wanted, thank you.

The failing test is a little worrisome. Please look into it a bit more. There might be a problem with how results are being found to compute marks - talk to @christinem about this, as she is working on basically the same problem.

**Pull request merged**